From a9e9e758646ee41289c0030645bbdd43c711e9c1 Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Fri, 18 May 2012 14:51:15 +0530
Subject: [PATCH 2/6] usb: musb: fix bug in data toggle sw workaround

Data toggle software workaround for extra IN issue introduced a bug for PIO mode
as data toggle is getting updated even for PIO mode which should not be done.

Merge to:
usb: musb: cppi41: correct data toggle mismatch to fix extra IN token issue
---
 drivers/usb/musb/musb_host.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/musb/musb_host.c b/drivers/usb/musb/musb_host.c
index 2195633..80d83bd 100644
--- a/drivers/usb/musb/musb_host.c
+++ b/drivers/usb/musb/musb_host.c
@@ -359,7 +359,8 @@ static inline void musb_save_toggle(struct musb_qh *qh, int is_in,
 		curr_toggle = csr ? 1 : 0;
 
 		/* check if data toggle has gone out of sync */
-		if (curr_toggle == qh->hw_ep->prev_toggle) {
+		if (is_dma_capable() && qh->hw_ep->rx_channel &&
+				curr_toggle == qh->hw_ep->prev_toggle) {
 			dev_dbg(musb->controller,
 				"Data toggle same as previous (=%d) on ep%d\n",
 					curr_toggle, qh->hw_ep->epnum);
-- 
1.7.7.6

